### ISSUE TITLE
Symmetrize edge vs node normalization

### ISSUE THREAD

**Issue by pgniewko** [pgniewko/gt-pyg#40]
## Summary

Edge features are systematically disadvantaged relative to node features due to normalization asymmetries at two levels:

1. **Model input level** Edge embeddings bypass `input_norm` and `input_dropout`
2. **GTConv layer level** Edge path uses post-norm while node path uses pre-norm

These compound across stacked layers, progressively dampening edge feature variance.

---

## Edge embeddings bypass input normalization and dropout

**File:** `gt_pyg/nn/model.py` (`GraphTransformerNet.forward()`)

### Current (asymmetric)
```
node_dim_in -> [node_emb] -> [input_norm] -> [input_dropout] -> GT layers
edge_dim_in -> [edge_emb] -------------------------------------> GT layers
```

### Proposed (symmetric)
```
node_dim_in -> [node_emb] -> [input_norm]    -> [input_dropout] -> GT layers
edge_dim_in -> [edge_emb] -> [edge_inp_norm] -> [input_dropout] -> GT layers
```

### Changes
- Add `self.edge_input_norm` (same type as `input_norm` -- BatchNorm1d or LayerNorm) in `__init__`
- Apply `self.edge_input_norm(e)` and `self.input_dropout(e)` after `self.edge_emb(edge_attr)` in `forward()`

---

## Edge post-norm vs node pre-norm asymmetry in GTConv

**File:** `gt_pyg/nn/gt_conv.py` (`GTConv.forward()`)

### Current (asymmetric)
```
NODES:  x_res -> norm1 -> Q,K,V -> attn -> WO -> drop -> (+x_res) -> norm2 -> FFN -> drop -> (+) -> x_out
                pre-norm                                               pre-norm

EDGES:  e_res -> (no norm) -> _eij -> WOe -> drop -> (+e_res) -> norm1e -> FFN_e -> drop -> (+) -> norm2e -> edge_out
                                                                 pre-norm                          POST-norm (extra!)
```

Edges have:
- No pre-norm before the attention output projection
- An extra post-norm (`norm2e`) on exit that nodes don't have

### Proposed (symmetric pre-norm)
```
NODES:  x_res -> norm1 -> Q,K,V -> attn -> WO -> drop -> (+x_res) -> norm2 -> FFN -> drop -> (+) -> x_out
                pre-norm                                               pre-norm

EDGES:  e_res -> norm1e -> _eij -> WOe -> drop -> (+e_res) -> norm2e -> FFN_e -> drop -> (+) -> edge_out
                pre-norm                                      pre-norm            (no post-norm)
```

### Changes
- Move `norm1e` to before the edge attention projection (pre-norm for attention, mirroring `norm1` for nodes)
- Rename current `norm1e` to `norm2e` (pre-FFN norm, mirroring `norm2` for nodes)
- Remove the final `norm2e` post-norm on edge output

---

## Files to modify
- `gt_pyg/nn/model.py` -- add `edge_input_norm`, apply norm + dropout to edge embeddings
- `gt_pyg/nn/gt_conv.py` -- restructure edge norm to match node pre-norm pattern

## Notes
- This is an architectural change that will affect trained model weights -- existing checkpoints will not be compatible
- Should be validated with training experiments to confirm improvement
